test/functional/admin/promotional_features_controller_test.rb

```diff
@@ -84,4 +84,43 @@ class Admin::PromotionalFeaturesControllerTest < ActionController::TestCase
     assert_not PromotionalFeature.exists?(promotional_feature.id)
     assert_equal "Promotional feature deleted.", flash[:notice]
   end
+
+  test "GET :reorder loads the promotional feature belonging to the organisation" do
+    promotional_feature1 = create(:promotional_feature, organisation: @organisation)
+    promotional_feature2 = create(:promotional_feature, organisation: @organisation)
+
+    get :reorder, params: { organisation_id: @organisation }
+
+    assert_response :success
+    assert_template :reorder
+    assert_equal [promotional_feature1, promotional_feature2], assigns(:promotional_features)
+  end
+
+  test "GET :reorder redirects to the promotional features index page when the org has less than 2 promotional features" do
+    create(:promotional_feature, organisation: @organisation)
+
+    get :reorder, params: { organisation_id: @organisation }
+
+    assert_redirected_to admin_organisation_promotional_features_path(@organisation)
+  end
+
+  test "PATCH on :update_order reorders the promotional_features and republishes the organisation" do
+    Organisation.stubs(:allowed_promotional).returns(organisations_array = mock)
+    organisations_array.expects(:find).with(@organisation.slug).returns(@organisation)
+    @organisation.expects(:reorder_promotional_features).with do |value|
+      value["2"] == "1" && value["1"] == "2"
+    end
+    Whitehall::PublishingApi.expects(:republish_async).with(@organisation)
```

Comment:
Just a suggestion (not even sure if it'll work)

Can we specify here that the Publishing API is **only called once**?

I'm thinking that'd be a nice way to lock-in the expectation that we don't end up republishing the organisation more than once. For example, it's protect against a future regression where somebody refactors the Publishing API call into an `after_save` callback on the ActiveRecord model.

However – it looks like you've stubbed out `Organisation#reoder_promotional_features` here, so actually this whole suggestion might be a moot point because that method isn't covered by this test.